libyul/optimiser/FullInliner.cpp

```diff
@@ -178,8 +179,17 @@ bool FullInliner::shallInline(FunctionCall const& _funCall, YulString _callSite)
 		return false;
 
 	// Do not inline into already big functions.
-	if (m_functionSizes.at(_callSite) > 45)
-		return false;
+	if (
```

Comment:
If we want that, we might have to write better stack-too-deep fixing algorithms first. I haven't checked with the latest version, but at some point the lower inlining limit actually *caused* stack too deep issues. The reason probably being around the fact that function calls fix the stack layouts, decreasing the flexibility of the stack layout generator.
The change is somewhat independent and I see that one might want it separate, but it's a bit tricky. It can't be done before this PR, since it then introduces stack-too-deep issues (we should probably actually verify that it still does, though), and it can't be done only after this PR, since with this PR it causes the kind of issues it originally was meant to avoid.
We should also consider how we want to deal with this in general in the future. There is little reason not to inline anything except that it *may* reduce code size (although in practice it actually does not seem to do so gravely or sometimes even the opposite happens).

Reply:
By the way: I was curious, so I applied this inlining change atop ``develop`` and compared gas costs for the tests that do still compile. The base branch is "develop + more inlining" compared to this PR.

<details><summary>Click for a table of gas differences</summary>

| File name                                                                |   IR-optimized (%) |   Legacy-Optimized (%) |   Legacy (%) |
|--------------------------------------------------------------------------|--------------------|------------------------|--------------|
| interface_inheritance_conversions.sol                                    |       -2.71593     |             0          |            0 |
| events/event_indexed_string.sol                                          |       -2.64654     |             0.00079624 |            0 |
| events/event_indexed_mixed.sol                                           |       -0.0548871   |             0          |            0 |
| events/event_dynamic_nested_array_storage_v2.sol                         |        0.0653524   |             0          |            0 |
| events/event_dynamic_array_storage_v2.sol                                |       -0.0597377   |             0          |            0 |
| events/event_dynamic_array_storage.sol                                   |       -0.0597377   |             0          |            0 |
| functionCall/failed_create.sol                                           |       -1.75784     |             0          |            0 |
| functionCall/mapping_array_internal_argument.sol                         |        0.00357309  |             0          |            0 |
| externalContracts/snark.sol                                              |       -9.45677     |             0          |            0 |
| externalContracts/ramanujan_pi.sol                                       |       -0.644357    |             0          |            0 |
| externalContracts/prbmath_unsigned.sol                                   |        1.17051     |             0          |            0 |
| externalContracts/deposit_contract.sol                                   |       -1.35501     |             0.182827   |            0 |
| externalContracts/prbmath_signed.sol                                     |        0.939367    |             0          |            0 |
| externalContracts/FixedFeeRegistrar.sol                                  |        0.39511     |             0          |            0 |
| externalContracts/strings.sol                                            |       -5.4438      |             0.43426    |            0 |
| constructor/arrays_in_constructors.sol                                   |        1.17685     |             0.673506   |            0 |
| constructor/no_callvalue_check.sol                                       |       -0.531988    |             0          |            0 |
| constructor/bytes_in_constructors_packer.sol                             |        3.05056     |             0          |            0 |
| abiencodedecode/abi_decode_simple_storage.sol                            |        0.0897026   |             0          |            0 |
| inlineAssembly/keccak_yul_optimization.sol                               |       -0.397887    |             0          |            0 |
| inheritance/inherited_function_calldata_calldata_interface.sol           |        0.501593    |             0          |            0 |
| inheritance/address_overload_resolution.sol                              |       -2.08835     |             0          |            0 |
| inheritance/inherited_function_calldata_memory_interface.sol             |        3.3913      |             0          |            0 |
| functionTypes/store_function.sol                                         |        0.253195    |             0          |            0 |
| storage/packed_storage_structs_bytes.sol                                 |        0.0492703   |             0          |            0 |
| abiEncoderV2/abi_encode_calldata_slice.sol                               |      -16.4417      |             0          |            0 |
| abiEncoderV2/storage_array_encoding.sol                                  |        0.06798     |             0          |            0 |
| abiEncoderV2/abi_encode_v2_in_modifier_used_in_v1_contract.sol           |       -2.40696     |             0          |            0 |
| abiEncoderV2/abi_encode_v2_in_function_inherited_in_v1_contract.sol      |       -0.683197    |             0.0259591  |            0 |
| abiEncoderV2/calldata_array.sol                                          |      -16.1769      |             0          |            0 |
| abiEncoderV2/abi_encode_v2.sol                                           |       -0.0494901   |             0          |            0 |
| abiEncoderV1/abi_encode_calldata_slice.sol                               |      -16.4417      |             0          |            0 |
| abiEncoderV1/abi_decode_v2_storage.sol                                   |        0.200321    |             0.0221609  |            0 |
| abiEncoderV1/struct/struct_storage_ptr.sol                               |       -0.0647435   |             0          |            0 |
| viaYul/array_storage_length_access.sol                                   |       -9.95961     |             0          |            0 |
| viaYul/array_storage_index_boundary_test.sol                             |       -8.1519      |             0          |            0 |
| viaYul/array_memory_index_access.sol                                     |      -14.2031      |             0          |            0 |
| viaYul/array_storage_index_access.sol                                    |       -0.789744    |             0          |            0 |
| viaYul/array_storage_push_empty_length_address.sol                       |       -8.9878      |             0          |            0 |
| viaYul/array_storage_push_empty.sol                                      |       -1.3157      |             0          |            0 |
| viaYul/array_storage_push_pop.sol                                        |       -3.21662     |             0          |            0 |
| viaYul/simple.sol                                                        |        0           |             0          |            0 |
| viaYul/array_storage_index_zeroed_test.sol                               |       -1.05594     |             0          |            0 |
| salted_create/salted_create.sol                                          |       -4.2666e-05  |             0          |            0 |
| salted_create/salted_create_with_value.sol                               |       -1.57419     |             0          |            0 |
| array/function_array_cross_calls.sol                                     |        4.43581     |             0.878437   |            0 |
| array/reusing_memory.sol                                                 |       -0.235616    |             0          |            0 |
| array/byte_array_transitional_2.sol                                      |       -4.71365     |             0          |            0 |
| array/bytes_length_member.sol                                            |        0.00181079  |             0          |            0 |
| array/dynamic_multi_array_cleanup.sol                                    |       -0.571536    |             0          |            0 |
| array/byte_array_storage_layout.sol                                      |       -0.613859    |             0          |            0 |
| array/dynamic_array_cleanup.sol                                          |       -0.110954    |             0          |            0 |
| array/fixed_arrays_as_return_type.sol                                    |       -1.82813     |             0.255718   |            0 |
| array/dynamic_arrays_in_storage.sol                                      |       -4.2433      |             0          |            0 |
| array/arrays_complex_from_and_to_storage.sol                             |        0.013696    |             0.108648   |            0 |
| array/fixed_array_cleanup.sol                                            |       -0.061182    |             0          |            0 |
| array/create_memory_array.sol                                            |       -9.63999     |             0          |            0 |
| array/delete/delete_storage_array_packed.sol                             |       -0.123035    |             0          |            0 |
| array/delete/bytes_delete_element.sol                                    |       -4.36959     |             0          |            0 |
| array/push/nested_bytes_push.sol                                         |       -0.194301    |             0          |            0 |
| array/push/array_push_packed_array.sol                                   |       -0.236584    |             0          |            0 |
| array/push/array_push_struct_from_calldata.sol                           |       -0.268887    |             0.0740645  |            0 |
| array/push/array_push_nested_from_calldata.sol                           |       -0.125539    |             0          |            0 |
| array/push/byte_array_push_transition.sol                                |       -4.91008     |             0          |            0 |
| array/push/push_no_args_2d.sol                                           |       -3.58309     |             0          |            0 |
| array/push/array_push_struct.sol                                         |       -0.0877828   |             0          |            0 |
| array/push/push_no_args_bytes.sol                                        |       -3.03789     |             0          |            0 |
| array/push/array_push.sol                                                |       -0.0621986   |             0          |            0 |
| array/copying/array_copy_storage_storage_different_base_nested.sol       |       -0.0356572   |             0          |            0 |
| array/copying/array_copy_storage_to_memory_nested.sol                    |        0.0609278   |             0          |            0 |
| array/copying/array_copy_storage_storage_static_static.sol               |       -0.091714    |             0          |            0 |
| array/copying/copy_function_storage_array.sol                            |       -0.141852    |             0          |            0 |
| array/copying/array_copy_different_packing.sol                           |       -0.0807497   |             0          |            0 |
| array/copying/array_copy_storage_storage_static_dynamic.sol              |       -0.0648819   |             0          |            0 |
| array/copying/array_copy_storage_storage_dynamic_dynamic.sol             |       -0.0598695   |             0          |            0 |
| array/copying/array_copy_clear_storage_packed.sol                        |       -0.0935223   |             0          |            0 |
| array/copying/array_copy_target_simple_2.sol                             |        0.00182874  |             0          |            0 |
| array/copying/array_nested_calldata_to_storage.sol                       |       -0.0927177   |             0          |            0 |
| array/copying/memory_dyn_2d_bytes_to_storage.sol                         |       -1.38337     |             0          |            0 |
| array/copying/array_storage_multi_items_per_slot.sol                     |       -0.014363    |             0          |            0 |
| array/copying/array_nested_memory_to_storage.sol                         |        0.100488    |             0          |            0 |
| array/copying/array_copy_including_array.sol                             |       -0.155978    |             0          |            0 |
| array/copying/array_copy_target_simple.sol                               |       -0.0387722   |             0          |            0 |
| array/copying/array_copy_cleanup_uint128.sol                             |       -0.091948    |             0          |            0 |
| array/copying/copy_byte_array_in_struct_to_storage.sol                   |        0.0406948   |             0          |            0 |
| array/copying/array_copy_target_leftover.sol                             |       -3.27056     |             0.690968   |            0 |
| array/copying/calldata_array_dynamic_to_storage.sol                      |       -0.00360013  |             0          |            0 |
| array/copying/array_copy_storage_storage_struct.sol                      |       -0.389178    |             0          |            0 |
| array/copying/bytes_inside_mappings.sol                                  |       -0.0101347   |             0          |            0 |
| array/copying/array_of_struct_memory_to_storage.sol                      |       -0.12563     |             0          |            0 |
| array/copying/copy_byte_array_to_storage.sol                             |       -0.312881    |             0          |            0 |
| array/copying/array_copy_calldata_storage.sol                            |        0.036389    |             0          |            0 |
| array/copying/copy_removes_bytes_data.sol                                |       -0.00394508  |             0          |            0 |
| array/copying/array_copy_target_leftover2.sol                            |        0.0216673   |             0          |            0 |
| array/copying/array_copy_nested_array.sol                                |        0.0205786   |             0          |            0 |
| array/copying/array_of_struct_calldata_to_storage.sol                    |       -0.131766    |             0          |            0 |
| array/copying/storage_memory_nested.sol                                  |       -0.167238    |             0          |            0 |
| array/copying/storage_memory_packed_dyn.sol                              |       -1.55384     |             0          |            0 |
| array/copying/array_of_structs_containing_arrays_memory_to_storage.sol   |       -0.00108842  |             0          |            0 |
| array/copying/array_copy_storage_storage_dyn_dyn.sol                     |       -0.0663659   |             0          |            0 |
| array/copying/storage_memory_nested_bytes.sol                            |        0.0478042   |             0          |            0 |
| array/copying/storage_memory_nested_struct.sol                           |       -0.156728    |             0          |            0 |
| array/copying/array_copy_storage_storage_different_base.sol              |       -0.0975774   |             0          |            0 |
| array/copying/array_copy_clear_storage.sol                               |       -0.0497808   |             0          |            0 |
| array/copying/array_of_structs_containing_arrays_calldata_to_storage.sol |       -0.408373    |             0          |            0 |
| array/copying/storage_memory_nested_from_pointer.sol                     |       -0.167238    |             0          |            0 |
| array/copying/arrays_from_and_to_storage.sol                             |       -0.273431    |             0.61589    |            0 |
| array/copying/bytes_storage_to_storage.sol                               |       -6.83751     |             0          |            0 |
| array/copying/array_copy_cleanup_uint40.sol                              |       -0.290945    |             0          |            0 |
| array/pop/array_pop_uint24_transition.sol                                |       -2.15989     |             0          |            0 |
| array/pop/byte_array_pop_masking_long.sol                                |       -0.754604    |             0          |            0 |
| array/pop/array_pop_uint16_transition.sol                                |       -3.02486     |             0          |            0 |
| array/pop/byte_array_pop_copy_long.sol                                   |       -0.418223    |             0          |            0 |
| array/pop/byte_array_pop_long_storage_empty_garbage_ref.sol              |       -1.97778     |             0          |            0 |
| array/pop/byte_array_pop_long_storage_empty.sol                          |       -4.18924     |             0          |            0 |
| array/pop/array_pop_array_transition.sol                                 |       -0.101577    |             0          |            0 |
| immutable/multi_creation.sol                                             |        0.211935    |             0          |            0 |
| structs/struct_copy_via_local.sol                                        |       -0.00729415  |             0          |            0 |
| structs/struct_delete_storage_nested_small.sol                           |       -0.18759     |             0          |            0 |
| structs/struct_delete_storage_with_array.sol                             |       -0.0833498   |             0          |            0 |
| structs/structs.sol                                                      |       -0.0222853   |             0          |            0 |
| structs/memory_structs_nested_load.sol                                   |       -0.0606357   |             0          |            0 |
| structs/struct_copy.sol                                                  |        0.0122447   |             0          |            0 |
| structs/struct_delete_storage_with_arrays_small.sol                      |       -0.0285684   |             0          |            0 |
| structs/struct_containing_bytes_copy_and_delete.sol                      |        0.0838292   |             0          |            0 |
| structs/struct_memory_to_storage_function_ptr.sol                        |       -0.0254116   |             0          |            0 |
| structs/conversion/recursive_storage_memory.sol                          |       -0.506612    |             0          |            0 |
| structs/calldata/calldata_struct_with_nested_array_to_storage.sol        |       -0.144487    |             0.122951   |            0 |
| various/staticcall_for_view_and_pure.sol                                 |       -2.53965e-06 |             0          |            0 |
| various/erc20.sol                                                        |       -2.7855      |             0          |            0 |
| various/destructuring_assignment.sol                                     |        0.0935884   |             0          |            0 |
| various/swap_in_storage_overwrite.sol                                    |       -0.0100272   |             0          |            0 |
| various/skip_dynamic_types_for_structs.sol                               |        0.0503449   |             0          |            0 |
</details>

So while there are some cases that are actually better off with the old code generator plus more inlining, the general tendency is still good (it'll be interesting to see if this improves further, once we fixed all the suboptimal placeholder algorithms in this PR).

But the following tests cannot even compile against develop with a higher inlining limit:

semanticTests/abiencodedecode/offset_overflow_in_array_decoding_2.sol
semanticTests/abiEncoderV2/dynamic_nested_arrays.sol
semanticTests/abiEncoderV2/memory_params_in_external_function.sol
semanticTests/abiEncoderV1/memory_params_in_external_function.sol
semanticTests/various/skip_dynamic_types_for_static_arrays_with_dynamic_elements.sol

But yeah - a major part of the gas savings in the full PR is of course due to this change :-). On the other hand the main motivation behind this entire effort was rather resilience against stack issues and better compatibility with stack-to-memory-moving anyways and gas savings are just a bonus.

Reply:
Oh, nice! If I see this correctly, the latest version of this PR in fact does no longer require this change. I'll confirm and, if possible, remove it.
This is the gas difference between develop and this PR without this change I just got, which is almost exclusively favourable as well:

<details><summary>Click for a table of gas differences</summary>

| File name                                                                |   IR-optimized (%) |   Legacy-Optimized (%) |   Legacy (%) |
|--------------------------------------------------------------------------|--------------------|------------------------|--------------|
| interface_inheritance_conversions.sol                                    |       -6.14832     |                0       |            0 |
| events/event_indexed_string.sol                                          |       -3.26668     |                0       |            0 |
| events/event_indexed_mixed.sol                                           |       -0.0548871   |                0       |            0 |
| events/event_dynamic_nested_array_storage_v2.sol                         |       -0.11565     |                0       |            0 |
| events/event_dynamic_array_storage_v2.sol                                |       -0.250844    |                0       |            0 |
| events/event_dynamic_array_storage.sol                                   |       -0.250844    |                0       |            0 |
| functionCall/failed_create.sol                                           |       -5.25951     |                0       |            0 |
| functionCall/mapping_array_internal_argument.sol                         |       -0.0793127   |                0       |            0 |
| externalContracts/snark.sol                                              |      -11.4843      |                0       |            0 |
| externalContracts/ramanujan_pi.sol                                       |       -6.34904     |               -3.87344 |            0 |
| externalContracts/prbmath_unsigned.sol                                   |       -0.443808    |               -1.33195 |            0 |
| externalContracts/deposit_contract.sol                                   |       -3.0234      |               -1.57194 |            0 |
| externalContracts/prbmath_signed.sol                                     |       -0.739431    |               -1.35104 |            0 |
| externalContracts/FixedFeeRegistrar.sol                                  |       -4.08363     |                0       |            0 |
| externalContracts/strings.sol                                            |      -10.7609      |               -2.88971 |            0 |
| constructor/arrays_in_constructors.sol                                   |       -1.7188      |                0       |            0 |
| constructor/no_callvalue_check.sol                                       |       -0.540637    |                0       |            0 |
| constructor/bytes_in_constructors_packer.sol                             |       -1.63406     |                0       |            0 |
| abiencodedecode/abi_decode_simple_storage.sol                            |       -0.00440428  |                0       |            0 |
| inlineAssembly/keccak_yul_optimization.sol                               |       -0.460011    |                0       |            0 |
| inheritance/inherited_function_calldata_calldata_interface.sol           |       -1.69397     |                0       |            0 |
| inheritance/address_overload_resolution.sol                              |       -4.3303      |                0       |            0 |
| inheritance/inherited_function_calldata_memory_interface.sol             |        2.97827     |                0       |            0 |
| functionTypes/store_function.sol                                         |        0.145923    |                0       |            0 |
| storage/packed_storage_structs_bytes.sol                                 |       -0.212866    |                0       |            0 |
| abiEncoderV2/abi_encode_calldata_slice.sol                               |      -16.6599      |                0       |            0 |
| abiEncoderV2/storage_array_encoding.sol                                  |        0.0217398   |                0       |            0 |
| abiEncoderV2/abi_encode_v2_in_modifier_used_in_v1_contract.sol           |       -2.42669     |                0       |            0 |
| abiEncoderV2/abi_encode_v2_in_function_inherited_in_v1_contract.sol      |       -2.97504     |                0       |            0 |
| abiEncoderV2/calldata_array.sol                                          |      -17.3535      |                0       |            0 |
| abiEncoderV2/abi_encode_v2.sol                                           |       -0.138103    |                0       |            0 |
| abiEncoderV1/abi_encode_calldata_slice.sol                               |      -16.6599      |                0       |            0 |
| abiEncoderV1/abi_decode_v2_storage.sol                                   |       -0.00392309  |                0       |            0 |
| abiEncoderV1/struct/struct_storage_ptr.sol                               |       -0.160524    |                0       |            0 |
| viaYul/array_storage_length_access.sol                                   |       -9.96014     |                0       |            0 |
| viaYul/array_storage_index_boundary_test.sol                             |       -7.95011     |                0       |            0 |
| viaYul/array_memory_index_access.sol                                     |      -14.1924      |                0       |            0 |
| viaYul/array_storage_index_access.sol                                    |       -1.33833     |                0       |            0 |
| viaYul/array_storage_push_empty_length_address.sol                       |       -8.98841     |                0       |            0 |
| viaYul/array_storage_push_empty.sol                                      |       -3.19377     |                0       |            0 |
| viaYul/array_storage_push_pop.sol                                        |       -3.21679     |                0       |            0 |
| viaYul/simple.sol                                                        |        0           |                0       |            0 |
| viaYul/array_storage_index_zeroed_test.sol                               |       -1.64137     |                0       |            0 |
| salted_create/salted_create.sol                                          |       -4.2666e-05  |                0       |            0 |
| salted_create/salted_create_with_value.sol                               |       -2.75672     |                0       |            0 |
| array/function_array_cross_calls.sol                                     |       -1.4636      |                0       |            0 |
| array/reusing_memory.sol                                                 |       -0.407834    |                0       |            0 |
| array/byte_array_transitional_2.sol                                      |       -5.37007     |                0       |            0 |
| array/bytes_length_member.sol                                            |       -0.0171837   |                0       |            0 |
| array/dynamic_multi_array_cleanup.sol                                    |       -1.46931     |                0       |            0 |
| array/byte_array_storage_layout.sol                                      |       -1.16149     |                0       |            0 |
| array/dynamic_array_cleanup.sol                                          |       -0.172745    |                0       |            0 |
| array/fixed_arrays_as_return_type.sol                                    |       -4.51293     |                0       |            0 |
| array/dynamic_arrays_in_storage.sol                                      |       -3.8181      |                0       |            0 |
| array/arrays_complex_from_and_to_storage.sol                             |       -0.36959     |                0       |            0 |
| array/fixed_array_cleanup.sol                                            |       -0.0613895   |                0       |            0 |
| array/create_memory_array.sol                                            |       -9.74456     |                0       |            0 |
| array/delete/delete_storage_array_packed.sol                             |       -0.271992    |                0       |            0 |
| array/delete/bytes_delete_element.sol                                    |       -4.36587     |                0       |            0 |
| array/push/nested_bytes_push.sol                                         |       -0.259903    |                0       |            0 |
| array/push/array_push_packed_array.sol                                   |       -0.465176    |                0       |            0 |
| array/push/array_push_struct_from_calldata.sol                           |       -0.324039    |                0       |            0 |
| array/push/array_push_nested_from_calldata.sol                           |       -0.260371    |                0       |            0 |
| array/push/byte_array_push_transition.sol                                |       -6.09131     |                0       |            0 |
| array/push/push_no_args_2d.sol                                           |       -3.61364     |                0       |            0 |
| array/push/array_push_struct.sol                                         |       -0.286811    |                0       |            0 |
| array/push/push_no_args_bytes.sol                                        |       -3.00625     |                0       |            0 |
| array/push/array_push.sol                                                |       -0.058417    |                0       |            0 |
| array/copying/array_copy_storage_storage_different_base_nested.sol       |       -0.0891284   |                0       |            0 |
| array/copying/array_copy_storage_to_memory_nested.sol                    |       -0.0333352   |                0       |            0 |
| array/copying/array_copy_storage_storage_static_static.sol               |       -0.0942296   |                0       |            0 |
| array/copying/copy_function_storage_array.sol                            |       -0.206498    |                0       |            0 |
| array/copying/array_copy_different_packing.sol                           |       -0.462161    |                0       |            0 |
| array/copying/array_copy_storage_storage_static_dynamic.sol              |       -0.066464    |                0       |            0 |
| array/copying/array_copy_storage_storage_dynamic_dynamic.sol             |       -0.083582    |                0       |            0 |
| array/copying/array_copy_clear_storage_packed.sol                        |       -0.346278    |                0       |            0 |
| array/copying/array_copy_target_simple_2.sol                             |       -0.116091    |                0       |            0 |
| array/copying/array_nested_calldata_to_storage.sol                       |       -0.287361    |                0       |            0 |
| array/copying/memory_dyn_2d_bytes_to_storage.sol                         |       -2.25024     |                0       |            0 |
| array/copying/array_storage_multi_items_per_slot.sol                     |       -0.132415    |                0       |            0 |
| array/copying/array_nested_memory_to_storage.sol                         |       -0.215609    |                0       |            0 |
| array/copying/array_copy_including_array.sol                             |       -0.167705    |                0       |            0 |
| array/copying/array_copy_target_simple.sol                               |       -0.174608    |                0       |            0 |
| array/copying/array_copy_cleanup_uint128.sol                             |       -0.380817    |                0       |            0 |
| array/copying/copy_byte_array_in_struct_to_storage.sol                   |        0.115222    |                0       |            0 |
| array/copying/array_copy_target_leftover.sol                             |       -5.47512     |                0       |            0 |
| array/copying/calldata_array_dynamic_to_storage.sol                      |       -0.0879839   |                0       |            0 |
| array/copying/array_copy_storage_storage_struct.sol                      |       -0.310951    |                0       |            0 |
| array/copying/bytes_inside_mappings.sol                                  |       -0.0259844   |                0       |            0 |
| array/copying/array_of_struct_memory_to_storage.sol                      |       -0.211694    |                0       |            0 |
| array/copying/copy_byte_array_to_storage.sol                             |       -0.50652     |                0       |            0 |
| array/copying/array_copy_calldata_storage.sol                            |       -0.0621341   |                0       |            0 |
| array/copying/copy_removes_bytes_data.sol                                |       -0.0157696   |                0       |            0 |
| array/copying/array_copy_target_leftover2.sol                            |       -1.00494     |                0       |            0 |
| array/copying/array_copy_nested_array.sol                                |       -0.0566493   |                0       |            0 |
| array/copying/array_of_struct_calldata_to_storage.sol                    |       -0.279925    |                0       |            0 |
| array/copying/storage_memory_nested.sol                                  |       -0.380186    |                0       |            0 |
| array/copying/storage_memory_packed_dyn.sol                              |       -0.724543    |                0       |            0 |
| array/copying/array_of_structs_containing_arrays_memory_to_storage.sol   |       -0.305819    |                0       |            0 |
| array/copying/array_copy_storage_storage_dyn_dyn.sol                     |       -0.0609521   |                0       |            0 |
| array/copying/storage_memory_nested_bytes.sol                            |        0.0457986   |                0       |            0 |
| array/copying/storage_memory_nested_struct.sol                           |       -0.272391    |                0       |            0 |
| array/copying/array_copy_storage_storage_different_base.sol              |       -0.139093    |                0       |            0 |
| array/copying/array_copy_clear_storage.sol                               |       -0.183019    |                0       |            0 |
| array/copying/array_of_structs_containing_arrays_calldata_to_storage.sol |       -1.00878     |                0       |            0 |
| array/copying/storage_memory_nested_from_pointer.sol                     |       -0.380186    |                0       |            0 |
| array/copying/arrays_from_and_to_storage.sol                             |       -0.357076    |                0       |            0 |
| array/copying/bytes_storage_to_storage.sol                               |       -9.04572     |                0       |            0 |
| array/copying/array_copy_cleanup_uint40.sol                              |       -1.06578     |                0       |            0 |
| array/pop/array_pop_uint24_transition.sol                                |       -2.45784     |                0       |            0 |
| array/pop/byte_array_pop_masking_long.sol                                |       -0.487721    |                0       |            0 |
| array/pop/array_pop_uint16_transition.sol                                |       -3.15571     |                0       |            0 |
| array/pop/byte_array_pop_copy_long.sol                                   |       -1.14996     |                0       |            0 |
| array/pop/byte_array_pop_long_storage_empty_garbage_ref.sol              |       -3.11104     |                0       |            0 |
| array/pop/byte_array_pop_long_storage_empty.sol                          |       -6.55097     |                0       |            0 |
| array/pop/array_pop_array_transition.sol                                 |       -0.153624    |                0       |            0 |
| immutable/multi_creation.sol                                             |        0.179098    |                0       |            0 |
| structs/struct_copy_via_local.sol                                        |       -0.021774    |                0       |            0 |
| structs/struct_delete_storage_nested_small.sol                           |       -0.220425    |                0       |            0 |
| structs/struct_delete_storage_with_array.sol                             |       -0.174862    |                0       |            0 |
| structs/structs.sol                                                      |       -0.0608042   |                0       |            0 |
| structs/memory_structs_nested_load.sol                                   |       -0.192081    |                0       |            0 |
| structs/struct_copy.sol                                                  |       -0.0227056   |                0       |            0 |
| structs/struct_delete_storage_with_arrays_small.sol                      |       -0.138418    |                0       |            0 |
| structs/struct_containing_bytes_copy_and_delete.sol                      |       -0.0604906   |                0       |            0 |
| structs/struct_memory_to_storage_function_ptr.sol                        |       -0.102536    |                0       |            0 |
| structs/conversion/recursive_storage_memory.sol                          |       -0.795715    |                0       |            0 |
| structs/calldata/calldata_struct_with_nested_array_to_storage.sol        |       -0.323436    |                0       |            0 |
| various/staticcall_for_view_and_pure.sol                                 |       -2.53965e-06 |                0       |            0 |
| various/erc20.sol                                                        |       -4.89774     |                0       |            0 |
| various/destructuring_assignment.sol                                     |       -0.0165475   |                0       |            0 |
| various/swap_in_storage_overwrite.sol                                    |       -0.0109335   |                0       |            0 |
| various/skip_dynamic_types_for_structs.sol                               |       -0.0122727   |                0       |            0 |
</details>

Reply:
Note: the balance stays positive even after https://github.com/ethereum/solidity/pull/10286